### ISSUE TITLE
The username you set in RetroArch should be the Citra Libretro username

### ISSUE THREAD

**Issue by hyarsan** [libretro/citra#12]
I feel like the username you set in RetroArch should be the Citra Libretro username. I would also recommend this be an core option that's on by default. It just makes sense to me! The same should happen with the language and keep in mind that **the language is not the same as the region**! RETRO_ENVIRONMENT_GET_USERNAME and RETRO_ENVIRONMENT_GET_LANGUAGE would need to be used.

![this_seems_logical_to_me](https://user-images.githubusercontent.com/32988993/32135167-dfa8ee0c-bbfa-11e7-81df-43633000a065.png)

This is an Libretro feature!

**Comment by j-selby** [comment]
This would require overwriting a user's NAND, so I'm not sure I am comfortable doing this in a automated fashion (might not be obvious to the user that we are modifying their files).

**Comment by hyarsan** [comment]
What about an core option that has two options, something like: "Use NAND System Settings" and "Use Frontend User Information". And when you choose the frontend option it would make an backup of the NAND system settings first and it would restore it if you switch back? Or is that too unreliable? Maybe make that the default and make sure it backs it up first?

**Comment by j-selby** [comment]
Too cumbersome I think. I'm thinking about adding some kind of additional menu system in the core that will provide a option to copy those settings (and handle multiplayer/etc).

**Comment by TheLocoBandito** [comment]
I don't know about this but the DeSmuME core have made something [similar](https://github.com/libretro/desmume/commit/d819b8addc0ca15e331e1a59736344d728add832) to what @hyarsan have originally suggested.

It works perfectly fine when you are about to start a new game in Zelda Phantom Hourglass or Spirit Tracks for example and your user name is showing up properly.

**Comment by hyarsan** [comment]
Yeah I noticed there was a commit for that. That's exactly what I wanted, I haven't tried it out yet.

**Comment by j-selby** [comment]
DS != 3DS. What works there may not work here.

**Comment by hyarsan** [comment]
Yes, but Libretro = Libretro still

**Comment by j-selby** [comment]
An abstract interface? Citra is totally different from, say, mgba. The interface only makes minimal assumptions about the internals about your core. This is the same here.

Desmume, from the looks of that commit, only stores those changes in memory, which is not possible here without changes to Citra's internals (which is outside of the design scope for citra-libretro).

**Comment by TheLocoBandito** [comment]
Yeah Citra is pretty different from DeSmuME, and what they have made to the DeSmuME core might not even be possible to implement for Citra but I thought like @hyarsan did with Libretro being Libretro and all.

A good idea in theory but probably a bad idea in practical terms.

**Comment by hyarsan** [comment]
https://cdn.discordapp.com/attachments/184109094070779904/421707829267660801/Lj9XS6w.png

It should be the RetroArch username instead of plain "LibRetro" :/

**Comment by j-selby** [comment]
If you were patient and wait until I *actually released something*, you would realise that that is exactly what is going on here.

Please don't jump to assumptions.

**Comment by hyarsan** [comment]
Well, I just wanted to make sure because that would be a good reason to close this issue.

**Comment by Papermanzero** [comment]
Although Desmume is completely different, I would also say that the options shall be used as template. How the core or emulator is doing that “internally” is secondary. In desmume the user can select in the core options the region and the language. This makes sense to have game options available. Only the username is taken from the account settings.

**Comment by j-selby** [comment]
@Papermanzero If you read the thread above, you would know that desmumes model won't work, period. 

Locking thread, as none of this is getting anywhere, and only serves to brigade. If you have actual helpful comments, ping me on discord. Thanks.
